### PR TITLE
Deny Clippy warnings and properly cancel previous runs of same workflow when commits are added

### PR DESCRIPTION
**Pull request by romac**
This setting cancels the previous runs of this workflow when a new commit is added to the PR, branch or tag:

```yaml
concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
  cancel-in-progress: true
```

### CODE REVIEW

File: .github/workflows/integration.yaml [hunk 1]
```diff
@@ -32,6 +32,10 @@ env:
   CARGO_NET_RETRY: 10
   RUSTUP_MAX_RETRIES: 10
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
```
Comment (mzabaluev) [inline .github/workflows/integration.yaml:38]: There is more going on here than just denying clippy warnings.

Can you describe what this change does?

Comment (romac) [reply .github/workflows/integration.yaml]: Updated the code + PR description.

File: .github/workflows/cargo-doc.yaml [hunk 2]
```diff
@@ -6,6 +6,11 @@ on:
       - master
   pull_request: {}
 
+# Cancel previous runs of this workflow when a new commit is added to the PR, branch or tag
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```
Comment (mzabaluev) [inline .github/workflows/cargo-doc.yaml:11]: So if not triggered on a `pull_request`, the expression `github.event.pull_request.number || github.ref` should fall back to the ref name and not just crash, right?

**Review by mzabaluev — APPROVED** [review]
Looks good to me.

File: .github/workflows/cargo-doc.yaml (hunk 2 above)
Comment (romac) [reply .github/workflows/cargo-doc.yaml:11]: Yes exactly

**Pull request merged**